src/Migrations/M032MigrateSocialMediaTwitterBlockToEmbedBlock.php

```diff
@@ -0,0 +1,360 @@
+                unset($block);
+
+                // Serialize the blocks content.
+                $new_content = serialize_blocks($blocks);
+
+                $post_update = array(
+                    'ID' => $post->ID,
+                    'post_content' => $new_content,
+                );
+
+                // Update the post with the replaced blocks.
+                $result = wp_update_post($post_update);
+
+                if ($result === 0) {
+                    throw new \Exception("There was an error trying to update the post #" . $post->ID);
+                }
+
+                echo "Migration successful\n";
+            }
+        } catch (\Exception $e) {
+            // Catch any exceptions and display the post ID if available
+            echo "Migration wasn't executed for post ID: ", $current_post_id ?? 'unknown', "\n";
+            echo $e->getMessage(), "\n";
+        }
+    }
+
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Get the social media URL from a Social Media block.
+     *
+     * @param array $block - A parsed Media block.
+     * @return string - The social media URL.
+     */
+    private static function get_social_media_url(array $block): mixed
+    {
+        $social_media_url = $block['attrs']['social_media_url'] ?? null;
+
+        return $social_media_url;
+    }
+
+    /**
+     * Get the source of a Social Media block.
+     * It can be Instagram, Facebook, or Twitter/X.
+     *
+     * @param string $url - The social media URL used to identify the source.
+     * @return string - The source of the Social Media block or an empty string if no source was found.
+     */
+    private static function identify_source(string $url): string
+    {
+        foreach (self::SOURCE_PATTERNS as $value => $key) {
+            if (preg_match($key, $url)) {
+                return $value;
+            }
+        }
+        return '';
+    }
+
+    /**
+     * Transform a Social Media block into a group of blocks.
+     * This group contains a header for the Social Media block title,
+     * a paragraph for the Social Media block description,
+     * and a final block (Embed) for the Twitter/X embed.
+     *
+     * @param array $block - A parsed Social Media block.
+     * @param string $source - The source of the Social Media block (Twitter or X).
+     * @param string $social_media_url - The Social Media URL.
+     * @return array - The transformed block.
+     */
+    private static function transform_blocks(array $block, string $social_media_url): array
+    {
+        $block_title = array_key_exists("title", $block['attrs']) ? $block['attrs']['title'] : null;
+        $block_description = array_key_exists("description", $block['attrs']) ? $block['attrs']['description'] : null;
+
+        $block = [];
+        $block['blockName'] = Utils\Constants::BLOCK_GROUP;
+        $block['attrs']['metadata']['name'] = 'Social Media Group';
```

Comment:
I think we can call this something like "Twitter" or "X", right?
Because we're only migrating Twitter elements.